packages/eas-cli/src/commands/build/version/set.ts

```diff
@@ -0,0 +1,188 @@
+  static hidden = true;
+
+  static flags = {
+    platform: Flags.enum({
+      char: 'p',
+      options: ['android', 'ios'],
+    }),
+    profile: Flags.string({
+      description:
+        'Name of the build profile from eas.json. Defaults to "production" if defined in eas.json.',
+      helpValue: 'PROFILE_NAME',
+    }),
+  };
+
+  public async runAsync(): Promise<void> {
+    const { flags } = await this.parse(BuildVersionSetView);
+
+    const projectDir = await findProjectRootAsync();
+
+    const platform = await selectPlatformAsync(flags.platform);
+    const easJsonReader = new EasJsonReader(projectDir);
+    await ensureRemoteVersionPolicyAsync(projectDir, easJsonReader);
+    const profile = await easJsonReader.getBuildProfileAsync(platform, flags.profile ?? undefined);
+
+    const exp = getExpoConfig(projectDir, { env: profile.env });
+    const projectId = await getProjectIdAsync(exp);
+    const projectFullName = await getProjectFullNameAsync(exp);
+    await validateAppConfigForRemoteVersionPolicyAsync(exp);
+
+    const applicationIdentifier = await getApplicationIdentifierAsync(
+      projectDir,
+      exp,
+      profile,
+      platform
+    );
+    const remoteVersions = await AppVersionQuery.latestVersionAsync(
+      projectId,
+      toAppPlatform(platform),
+      applicationIdentifier
+    );
+    const currentStateMessage = remoteVersions?.buildVersion
+      ? `Project ${chalk.bold(projectFullName)} with ${getApplicationIdentifierName(
+          platform
+        )} "${applicationIdentifier}" is configured with ${getBuildVersionName(platform)} ${
+          remoteVersions.buildVersion
+        }.`
+      : `Project ${chalk.bold(projectFullName)} with ${getApplicationIdentifierName(
+          platform
+        )} "${applicationIdentifier}" does not have any ${getBuildVersionName(
+          platform
+        )} configured.`;
+
+    const versionPromptMessage = remoteVersions?.buildVersion
+      ? `What version would you like to set?`
+      : `What version would you like to initialize it with?`;
+    Log.log(currentStateMessage);
+
+    const { version } = await promptAsync({
+      type: platform === Platform.ANDROID ? 'number' : 'text',
+      name: 'version',
+      message: versionPromptMessage,
+      validate:
+        platform === Platform.ANDROID
+          ? value => isValidVersionCode(value) || INVALID_VERSION_CODE_ERROR_MESSAGE
+          : value => isValidBuildNumber(value) || INVALID_BUILD_NUMBER_ERROR_MESSAGE,
+    });
+    await AppVersionMutation.createAppVersionAsync({
+      appId: projectId,
+      platform: toAppPlatform(platform),
+      applicationIdentifier,
+      storeVersion: exp.version ?? '1.0.0',
+      buildVersion: String(version),
+      runtimeVersion: getRuntimeVersionNullable(exp, platform) ?? undefined,
+    });
+  }
+}
+
+function getApplicationIdentifierName(platform: Platform): string {
+  if (platform === Platform.ANDROID) {
+    return 'applicationId';
```

Comment:
```suggestion
    return 'application ID';
```

this wording is preferred in a sentence, eg: https://developer.android.com/studio/build/configure-app-module